.github/workflows/release-publish.yml

```diff
@@ -0,0 +1,57 @@
+name: 'Release: Publish'
+
+on:
+  pull_request:
+    types:
+      - closed
+    branches:
+      - 'release/patch'
+      - 'release/minor'
+
+jobs:
+  publish-release:
+    if: github.event.pull_request.merged == true
+    runs-on: ubuntu-latest
+
+    permissions:
+      contents: write
+
+    timeout-minutes: 10
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - uses: pnpm/action-setup@v2.2.4
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 16.x
+          cache: 'pnpm'
+      - run: pnpm install --frozen-lockfile
+
+      - name: Build
+        run: pnpm build
+
+      - name: Publish to NPM
+        run: |
+          echo "//registry.npmjs.org/:_authToken=${{ secrets.NPM_TOKEN }}" > ~/.npmrc
```

Comment:
One of the admins need to add `NPM_TOKEN` to [our repo secrets](https://github.com/n8n-io/n8n/settings/secrets/actions).